pkg/controller/instance/instance_controller.go

```diff
@@ -316,7 +308,62 @@ func instanceEventFilter(mgr manager.Manager) predicate.Funcs {
 	}
 }
 
-func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Instance) error {
+func createPlan(r *ReconcileInstance, planName string, instance *kudov1alpha1.Instance) error {
+	ctx := context.TODO()
+	gvk, _ := apiutil.GVKForObject(instance, r.scheme)
+
+	planExecution := getPlanExecution(gvk, instance, planName)
```

Comment:
this method feels like it's getting something while it's actually creating new plan execution. Can we maybe find better name for that?

Reply:
maybe `newPlanExecution`?

Reply:
good point... I actually fine inclined to move this to planexecution namespace... please hold.